### PR TITLE
feat(manager/pip-compile): Add extractHeaderCommand method

### PR DESCRIPTION
**Pull request by not7cd**
## Changes

This adds `extractHeaderCommand()` method needed for `extractAllPackageFiles()`. There is a code duplication within `contructPipCompileCmd()`, but this will be refactored in the next PRs. 
The method itself shouldn't change the extracted command, but give all information it found.

## Context

#26858 too big

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/pip-compile/common.ts [hunk 1]
```diff
@@ -56,11 +59,158 @@ export async function getExecOptions(
...
+        result.extra.push(value);
+      } else if (option === '--extra-index-url') {
+        result.extraIndexUrl = result.extraIndexUrl ?? [];
+        result.extraIndexUrl.push(value);
+        // TODO: add to secrets? next PR
+      } else if (option === '--constraint') {
+        result.constraintsFiles = result.constraintsFiles ?? [];
+        result.constraintsFiles.push(value);
+      } else if (option === '--output-file') {
+        result.outputFile = value;
```
Comment (viceice) [inline lib/modules/manager/pip-compile/common.ts:153]: should we check for duplicates of this? we don't support that

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/pip-compile/common.ts [hunk 2]
```diff
@@ -56,11 +59,158 @@ export async function getExecOptions(
...
+  '--index-url',
+  ...optionsWithArguments,
 ];
+
+// TODO(not7cd): test on all correct headers, even with CUSTOM_COMPILE_COMMAND
+export function extractHeaderCommand(
+  content: string,
+  fileName: string,
+): PipCompileArgs {
+  const strict: boolean = true; // TODO(not7cd): add to function params
```
[the remark this answers is not in the document]
Comment (not7cd) [reply lib/modules/manager/pip-compile/common.ts:98]: Removed

**Comment by not7cd** [comment]
not sure what is wrong with tests, as patch coverage is 100%

**Review by viceice — APPROVED** [review]

**Pull request merged**